app/routes/datasets/queries.py

```diff
@@ -155,6 +160,9 @@ async def query_dataset_json(
     """
 
     dataset, version = dataset_version
+    if dataset in PROTECTED_QUERY_DATASETS:
+        await is_gfwpro_admin(error_str="Unauthorized query on a restricted dataset")
```

Comment:
Similar to above comment, do we really want to get in the business of restricting hardcoded datasets to hardcoded users? Seems like it goes against the general design of the API, and we'll need to redeploy every time we need to add new datasets or users to these lists.

Long term solution seems more like restricted datasets should only be accessible by the admin user that uploaded it.

But once again, that's pretty complicated, and this situation has only come up once. So ok with this as a stopgap until we start seeing this more regularly from Pro or other users.